cmake/app/boilerplate.cmake

```diff
@@ -347,3 +347,11 @@ See http://docs.zephyrproject.org/latest/porting/board_porting.html#old-zephyr-k
 To get rid of this note, create a file called 'hide-defaults-note' in the \n\
 Zephyr root directory. An empty file is fine.")
 endif()
+
+if("${CMAKE_EXTRA_GENERATOR}" STREQUAL "Eclipse CDT4")
+  # Call the amendment function before .project and .cproject generation
+  # C and CXX includes, defines in .cproject without __cplusplus
+  # with project includes and defines
+  include(${ZEPHYR_BASE}/cmake/util/eclipse_cdt4_generator_amendment.cmake)
```

Comment:
util is reserved for standalone CMake "scripts", this belongs in zephyr/cmake/ somewhere.